### ISSUE TITLE
Frequent "OSError" triggered by `exception_hook` using Honeybadger for Python 0.12.0 

### ISSUE THREAD

**Issue by agdiaz** [honeybadger-io/honeybadger-python#121]
Hello Honeybadger devs,
I hope this message finds you doing well! 

I've implemented the error tracking service on a Django website deployed on "Ubuntu 18.04.3 LTS". This application runs by using a `uwsgi` engine and behind an NGINX server. 

Several times a day I received notifications related to an error coming from inside the HoneyBadger library but with an empty stack trace:

![image](https://user-images.githubusercontent.com/1646576/195134760-486626f1-fbd9-4c16-981e-8480a4700174.png)

There are no further details that could help me to identify the issue inside my project's code:
![image](https://user-images.githubusercontent.com/1646576/195134530-c020e6fa-d6b4-45aa-b29d-2768ec7cf51f.png)

Would you mind helping me to identify the root cause of this issue? 

Thank you in advance for your feedback and time.

Best from Brussels!
Adrián,

**Comment by joshuap** [comment]
@Kelvin4664 can you look at this one? Could it be a case where the stack trace is being generated internally when calling `honeybadger.notify` like this example from the README? 

```python
honeybadger.notify(error_class='ValueError', error_message='Something bad happened!', fingerprint='custom_fingerprint')
```

**Comment by agdiaz** [comment]
Hello there, 

Thanks for checking it out! Just to add a bit more context, I'm attaching the frequency chart in order to illustrate that it's happening randomly: 

![image](https://user-images.githubusercontent.com/1646576/195314251-3102a85c-e28f-4676-8201-577365b8807d.png)

I've reviewed the source code of my project, and the two places where I invoke `honeybadger.notify` add context using a dictionary, so these places don't seem to be related to the `OSError` because we should see the context at least.

Have a nice day 👋

**Comment by Kelvin4664** [comment]
Hi @agdiaz
Thank you for the detailed report.
There is little or no detail because the error occurred outside your application. More specifically, at the `uwsgi` layer which outside honeybadger's frame.
Such errors usually occur when uwsgi encounters a large request body, or having more requests than there's available workers.
You can turn these notifications off from your uwsgi config by turning on the following:
`ignore-sigpipe`
`ignore-write-errors`
`disable-write-exception`

@joshuap can we consider a dedicated wsgi/uwsgi intergration to help with cases like this?

**Comment by joshuap** [comment]
> @joshuap can we consider a dedicated wsgi/uwsgi intergration to help with cases like this?

Yep, could you write up an issue for that feature and give it the "enhancement" label? Thanks!

**Comment by agdiaz** [comment]
@Kelvin4664 @joshuap: thank you for your help finding the root cause of these `OSError`s and for proposing a workaround!

**Comment by Kelvin4664** [comment]
Will be addressed with #125
